### PR TITLE
Update the interface spec to describe the limit on threshold ECDSA derivation paths

### PR DESCRIPTION
**Pull request by randombit**
BIP32 allows a derivation depth of at most 255, and we enforce this limit.

FOLLOW-1013

### CODE REVIEW

**Comment by robin-kunzler** [comment]
Thanks for addressing this @randombit ! Can't approve but LGTM :-)

**Comment by randombit** [comment]
Ping @Dfinity-Bjoern

File: spec/index.adoc [hunk 1]
```diff
@@ -1681,7 +1681,7 @@ This method takes no input and returns 32 pseudo-random bytes to the caller. The
 NOTE: The ECDSA API is considered EXPERIMENTAL. Canister developers must be aware that the API may evolve in a non-backward-compatible way.
 
 This method returns a https://www.secg.org/sec1-v2.pdf[SEC1] encoded ECDSA public key for the given canister using the given derivation path. If the `canister_id` is unspecified, it will default to the canister id of the caller.
-The `derivation_path` is a vector of variable length byte strings.
+The `derivation_path` is a vector of variable length byte strings, containing at most 255 byte strings.
```
Comment (Dfinity-Bjoern) [inline spec/index.adoc:1684]: Is 255 a bound on the length of the vector or a bound on the length of the strings in the vector?

Comment (randombit) [reply spec/index.adoc]: It is the bound on the length of the vector; the length of each individual string is arbitrary. I amended the comment, PTAL.

Comment (mraszyk) [reply spec/index.adoc]: Thank you!

**Pull request merged**